indexing-service/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorResource.java

```diff
@@ -395,6 +395,25 @@ public Response shutdown(@PathParam("id") final String id)
     return terminate(id);
   }
 
+  @POST
+  @Path("/{id}/taskGroups/restart")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(SupervisorResourceFilter.class)
+  public Response handoffTaskGroups(@PathParam("id") final String id, final List<Integer> taskGroupIds)
+  {
+    return asLeaderWithSupervisorManager(
+        manager -> {
+          if (manager.handoffTaskGroupsEarly(id, taskGroupIds)) {
```

Comment:
Should handle error case when the supervisor does not support handoffTaskGroupsEarly

We probably also need a null or empty check for taskGroupIds.